pkg/fission-cli/cmd/spec/validate.go

```diff
@@ -56,11 +57,16 @@ func (opts *ValidateSubCommand) run(input cli.Input) error {
 	}
 
 	// this does the rest of the checks, like dangling refs
-	err = fr.Validate(input)
+	warning, err := fr.Validate(input)
 	if err != nil {
 		return errors.Wrap(err, "error validating specs")
 	}
-
+	if warning {
```

Comment:
We don't need this if-else actually. We can always print `Spec validation successful\nSpec contains ....` and shows warnings in the end.

Reply:
Okay. We can use the warning list here.